File: src/main/java/composing/methods/Food.java
package composing.methods;

import java.time.LocalDate;

public class Food {
    private LocalDate expirationDate;
    private Boolean approvedForConsumption;
    private Integer inspectorId;

    public Food(LocalDate expirationDate, Boolean approvedForConsumption, Integer inspectorId) {
        this.expirationDate = expirationDate;
        this.approvedForConsumption = approvedForConsumption;
        this.inspectorId = inspectorId;
    }

    public boolean isEdible() {
        if (this.expirationDate.isAfter(LocalDate.now()) && this.approvedForConsumption == true && this.inspectorId != null) {
            return true;
        } else {
            return false;
        }
    }
}

File: src/scala/src/main/scala/composing/methods/Food.scala
package composing.methods

import java.time.LocalDate

final case class Food(
    expirationDate: LocalDate,
    approvedForConsumption: Boolean,
    inspectorId: Option[Int]
) {
  lazy val isEdible: Boolean =
    isFresh && approvedForConsumption && hasBeenInspected
  val isFresh: Boolean = expirationDate.isAfter(LocalDate.now)
  val hasBeenInspected: Boolean = inspectorId.isDefined
}
